lib/nx_zlib.c

```diff
@@ -468,34 +499,49 @@ int nx_submit_job(nx_dde_t *src, nx_dde_t *dst, nx_gzip_crb_cpb_t *cmdp, void *h
 	return cc;
 }
 
+static inline void nx_device_free(nx_devp_t nx_devp) {
+	nx_function_end(nx_devp->vas_handle);
+	free(nx_devp);
+}
+
 nx_devp_t nx_open(int nx_id)
 {
-	nx_devp_t nx_devp;
+	nx_devp_t nx_devp, saved;
 	void *vas_handle;
 	pid_t my_pid;
-	int ocount;
+
+	my_pid = getpid();
 
 	if (disable_saved_nx_devp == 0) {
+		pthread_mutex_lock(&saved_nx_devp_mutex);
```

Comment:
There are 2 returns from this function that may happen without unlocking this mutex.
Is this intentional?

Reply:
Good point. It's not. I'll fix it.